agent_build_refactored/managed_packages/steps/prepare_python_environment.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2014-2022 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is meant to be executed by the instance of the 'agent_build_refactored.tools.runner.RunnerStep' class.
+# Every RunnerStep provides common environment variables to its script:
+#   SOURCE_ROOT: Path to the projects root.
+#   STEP_OUTPUT_PATH: Path to the step's output directory.
+#
+# This script prepares toolset environment that will be used during packages build. For example, it installs
+# the fpm command line tools which is used to create deb and rpm packages.
+#
+# It expects next environment variables:
+#   BUILD_PYTHON: output path of the previous step that provides Python interpreter.
+#   BUILD_AGENT_LIBS: output path of the previous step that provides dev libraries for the Python.
+#   FPM_VERSION: Version of the fpm tool.
+#   PACKAGE_CLOUD_VERSION: Version of the package_cloud tools. used to manipulate Packagecloud packages.
+#
+
+set -e
+
+cp -a "${BUILD_PYTHON_1_1_1}/." /
+cp -a "${BUILD_OPENSSL_1_1_1}/." /
+cp -a "${BUILD_DEV_REQUIREMENTS}/root/." /
+
+echo "${PYTHON_INSTALL_PREFIX}/lib" >> /etc/ld.so.conf.d/python3.conf
+ldconfig
+
+ln -s "${PYTHON_INSTALL_PREFIX}/bin/python3" /usr/bin/python3
+
+# shellcheck disable=SC1090
+source ~/.bashrc
```

Comment:
Sometimes there may be stuff that has to be activated first in order to make it work in the current bash script. For example, in order to prepare the environment to build our embedded Python interpreter ([link](https://github.com/scalyr/scalyr-agent-2/blob/3a02ca9d48f6678a6045479164679ac1f29fceba/agent_build_refactored/managed_packages/steps/install_build_environment/install_gcc_centos_7.sh#L30)) , we install a special package `devtoolset-9`  that provides more recent version of the `gcc`, but this can be used only after "enabling" it.  Unfortunately, just putting this into the `.bashrc`  is not enough in our case, because our "step" scripts are executed in docker by using `docker run ... bash ...` command, and bash, in this case, does not automatically invoke the `.bashrc`.